libp2p/switch.nim

```diff
@@ -257,6 +257,45 @@ proc upgradeIncoming(s: Switch, incomingConn: Connection) {.async, gcsafe.} = #
   finally:
     await incomingConn.close()
 
+proc dialAndUpgrade(s: Switch,
+                    peerId: PeerID,
+                    addrs: seq[MultiAddress]):
+                    Future[Connection] {.async.} =
+  debug "Dialing peer", peerId
+  for t in s.transports: # for each transport
+    for a in addrs: # for each address
+      if t.handles(a): # check if it can dial it
+        trace "Dialing address", address = $a, peerId
+        let dialed = try:
+            await t.dial(a)
+          except CancelledError as exc:
+            debug "Dialing canceled", msg = exc.msg, peerId
+            raise exc
+          except CatchableError as exc:
+            debug "Dialing failed", msg = exc.msg, peerId
+            libp2p_failed_dials.inc()
+            continue # Try the next address
```

Comment:
won't this continue cause issues?

Reply:
it doesn't have any finally in that block anymore or will this also cause issues?

Reply:
but it's also not needed anymore

Reply:
ok, but the question still stands, would it cause issue or is it unknown at this point?

Reply:
right, it was finally blocks that got skipped in this case - maybe works, but a _test_ would confirm it

Reply:
Hmm, yeah a test would be good - I think we can add one now that we have `accept` and control over the incoming flow.